xenmgr/XenMgr/HostOps.hs

```diff
@@ -47,6 +48,9 @@ import XenMgr.Config
 hostShutdown :: XM ()
 hostShutdown = (hostWhenIdleDoWithState HostShuttingDown $ executePmAction ActionShutdown) >> return ()
 
+hostShutdownIdle :: XM ()
+hostShutdownIdle = executePmAction ActionIdleShutdown >> return ()
```

Comment:
Above and below, the actions are prefixed with hostWhenIdleDoWithState.  Is that intentional?  I can see it being intentional since when ShutdownIdle is called, we don't want to wait on something else.

Reply:
> Is that intentional?

Should be "Is not using hostWhenIdleDoWithState intentional?"

Reply:
The ```HostIdle``` xenmgr-ism is only set when going into host S3 (```execute_ ActionSleep``` on line 345).  ```hostWhenIdleDoWithState``` just does the action if the Host State is ```host-idle```, updates the state, then immediately sets the state back to ```host-idle```.  

My guess the history here was some automatic Sleep request after being idle, but the current Idle system is now managed in the input part of Glass and xcpmd.  We don't key off ```HostState``` for anything in this stack, so I'm not using the ```hostWhenIdleDoWithState``` method.